crates/cli-support/src/js/binding.rs

```diff
@@ -783,6 +787,45 @@ fn instruction(js: &mut JsBuilder, instr: &Instruction, log_error: &mut bool) ->
             js.push(format!("len{}", i));
         }
 
+        Instruction::UnwrapResult { table_and_alloc } => {
+            // The top of the stack is a nullable u32. If it is nonzero, takeObject and throw it.
+            let i = js.tmp();
+            let j = js.tmp();
+            // Err is popped first. The original layout was: ResultAbi {
+            //    abi: ResultAbiUnion<T>,
+            //    is_ok: u32,
+            //    err: u32,
+            // }
+            let err = js.pop();
+            let is_ok = js.pop();
+            js.prelude(&format!("var is_ok{i} = {is_ok};", i = i, is_ok = is_ok));
+            js.prelude(&format!("var err{j} = {err};", j = j, err = err));
+            if let Some((table, alloc)) = table_and_alloc {
+                let dereference = js.cx.expose_get_from_externref_table(*table, *alloc)?;
+                js.prelude(&format!(
+                    "
+                    if (is_ok{i} === 0) {{
+                        throw {}(err{j});
+                    }}
+                    ",
+                    dereference,
+                    i = i,
+                    j = j,
+                ));
+            } else {
+                js.cx.expose_take_object();
+                js.prelude(&format!(
+                    "
+                    if (is_ok{i} === 0) {{
+                        throw takeObject(err{j});
```

Comment:
I'm a bit confused on why one branch reaches into a table and one branch uses `takeObject`, presumably the branch that reaches into the table needs to also "deallocate" the slot in the table, right? I would naively expect both branches here to use `takeObject` or otherwise a slot of the table is leaked.

Reply:
Aha ok rereading your comments I see you are also as stumped about this as I am. I think the bug is that we need to deallocate the index in the `externref` case. I believe that the `__externref_table_dealloc` intrinsic defined in `src/externref.rs` is what needs to be invoked after the table slot is loaded from. I imagine if you poke around the various bits and pieces of the JS glue code generation for other usages of that I think it should be relatively easy to pattern-match on what is already done and doing roughly the same thing here too.

Reply:
Yep, this worked nicely.